// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option go_package = "comtrade-gprc/comtrade";
option java_multiple_files = true;
option java_package = "io.grpc.examples.comtradegrpc";
option java_outer_classname = "ComtradeGRPCProto";

package comtrade;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc FastLoad (DatParseParam) returns (WaveDataReply){}
  rpc TestFastLoad (HelloRequest) returns (Analog){}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message AnalogChannel{
  repeated double value = 1;
}

message Analog{
  repeated AnalogChannel channel = 1;
}

message StatusChannel{
  repeated int32 value=1;
}

message Status{
  repeated StatusChannel channel=1;
}

message Time{
  repeated double value=1;
}

message WaveDataReply{
  Analog analog=1;
  Status status=2;
  Time t = 3;
}

message SampleRateRow{
  repeated int32 sampleRateRow = 1;
}

message DatParseParam{
  string datFile = 1;
  int32 timeMult = 2;
  double timeBase = 3;
  int32 achannels = 4;
  int32 schannels =5;
  repeated double a =6;
  repeated double b = 7;
  int32 groupsOf16Bits = 8;
  int32 totalSample = 9 ;
  bool timestampCritical = 10;
  repeated SampleRateRow sampleRate = 11;
}